### PR TITLE
H264frame fix

### PR DESCRIPTION
**Pull request by imbcmdth**
There are cases when nalUnit streams may not start with an AUD and this can result in `h264Frame` being undefined as we parse more

### CODE REVIEW

File: lib/flv/transmuxer.js [hunk 1]
```diff
@@ -227,26 +227,35 @@ VideoSegmentStream = function(track) {
...
+        config = currentNal.config;
+        track.width = config.width;
+        track.height = config.height;
+        track.sps = [currentNal.data];
+        track.profileIdc = config.profileIdc;
+        track.levelIdc = config.levelIdc;
+        track.profileCompatibility = config.profileCompatibility;
+        h264Frame.endNalUnit();
+      } else if (currentNal.nalUnitType === 'pic_parameter_set_rbsp') {
+        track.newMetadata = true;
```
Comment (dmlap) [inline lib/flv/transmuxer.js]: Tangential to this review but why is this boolean necessary? Wouldn't `track.pps.length` suffice?

File: lib/flv/transmuxer.js [hunk 2]
```diff
@@ -227,26 +227,35 @@ VideoSegmentStream = function(track) {
       return;
     }
 
+    // Throw away nalUnits at the start of the byte stream until we find
+    // the first AUD
```
Comment (dmlap) [inline lib/flv/transmuxer.js]: This step is not necessary for the MP4 transmuxer?

**Comment by dmlap** [comment]
A couple questions but possibly no action necessary. LGTM

File: lib/flv/transmuxer.js (hunk 1 above)
Comment (imbcmdth) [inline lib/flv/transmuxer.js]: Unfortunately, `track.[pps/sps]` is simply set every time we see a pps or sps so except for the first time it'll always be 1. We need to emit the metadata tags for based on sps and pps every keyframe so we need to keep the data around.

The boolean just signals that the metadata *may* have changed so emit it even if we *aren't* at a keyframe.

File: lib/flv/transmuxer.js (hunk 2 above)
Comment (imbcmdth) [inline lib/flv/transmuxer.js]: I can't think of a good reason NOT to do this..

**Pull request merged**